proposed/activity-improvements.md

```diff
@@ -0,0 +1,290 @@
+
+Today distributed tracing instrumentation is predominantly only added to a few critical libraries such as ASP.Net's Kestrel web server or the HttpClient implementation in the BCL. Even if other libraries or app developers wanted to add this instrumentation to their code, the current design makes it difficult - we want to fix that.  We have also been watching the [OpenTelemetry](https://opentelemetry.io/) project which is defining a multi-language multi-vendor standard for how distributed tracing should work. Our goal is to ensure that .Net integrates smoothly. For app developers this means that you can add a reference to the OpenTelemetry NuGet package and all the Activity-based instrumentation in your codebase, your 3rd party libraries, and the runtime will light up and get collected. For library authors this means that you can write to a platform standard API that doesn't force any specific dependency on the library users, but lights up in a predictable way when OpenTelemetry (or any other agent) is present.  
+
+We expect to achieve this by additions to the Activity object model and a new ActivityListener type. These additions should provide stronger contracts between in-proc producers and consumers of distributed tracing data, a richer set of information that Activity can represent, and streamlined calling patterns for typical scenarios.
+
+## Scenarios and User Experience
+
+#### 1. A developer wants a portion of code to be annotated with a named block in a distributed trace Gantt chart
+
+The developer needs to wrap their region of code in a using statement that creates the Activity and assigns it a name. It is possible that the activity reference will be null if there is nothing in the process listening for this telemetry or if only a sub-set of telemetry is being sampled.
+
+```C#
+    using (Activity? activity = Activity.StartNew("MyLibrary.Subcomponent.FooOperation")
+    {
+        // optionally the activity can be enriched with extra data
+        activity?.AddTag("userId", userId);
+        activity?.AddTag("fooParam", foo);
+            
+        // the developer's original code goes here
+    }
+```
+
+### 2. Monitoring agent wants to listen to activities
+
+A monitoring agent can create a listener which represents their subscription to activity callbacks like this.
+```C#
+
+IDisposable listener = Activity.StartListening(
```

Comment:
`Tracer` in OpenTelemetry allows to add attributes and do other modification on the `Activity` to be created. This doesn't seem to be achievable with this design.

Also semantic of `shouldCreateActivity` is not declared. Ideally it must be three-fold. No activity (disabled), Activity with context propagation, but no recording. And Activity that is being actively recorded so all attributes needs to be populated on it